src/fractal_ome_zarr_hcs_stitching/stitching_task.py

```diff
@@ -85,78 +85,114 @@ def stitching_task(
         f"{ngff_image_meta.get_pixel_sizes_zyx(level=1)}"
     )
 
-    # Load FOVs for registration
-    xim_well = get_sim_from_multiscales(
-        Path(zarr_url), 0
-    )  # could also be lower resolution
     fov_roi_table = ad.read_zarr(Path(zarr_url) / "tables/FOV_ROI_table").to_df()
-    msims = get_tiles_from_sim(xim_well, fov_roi_table)
-
-    reg_spatial_dims = [
-        dim
-        for dim in msi_utils.get_sim_from_msim(msims[0]).dims
-        if dim in ["z", "y", "x"]
-    ]
+    input_transform_key = "fractal_input"
 
     #############
     # Registration
     ##############
 
+    # Load FOVs for registration
+    xim_well_reg = get_sim_from_multiscales(
+        Path(zarr_url), resolution=registration_resolution_level
+    )  # could also be lower resolution
+
+    input_spatial_dims = si_utils.get_spatial_dims_from_sim(
+        xim_well_reg.squeeze(drop=True)
+    )
+
+    # determine whether to perform registration on maximum projection in Z
+    reg_max_project_z = registration_on_z_proj and ("z" in input_spatial_dims)
+
+    if reg_max_project_z:
+        xim_well_reg = xim_well_reg.max("z")
+    else:
```

Comment:
Pure style comment, but you can drop this else, it has no effect :)

Reply:
(i.e. drop the else and the line after it)

Reply:
Oh right, well spotted 🙏 This is a leftover from some renaming 😁